### PR TITLE
IOS Mute button

### PR DESCRIPTION
**Pull request by MarekSabol**
Hi, I've developed it based on our discussion, hopefully you will be satisfied with the implementation :). In case there will be some changes needed, let me know :).

Snímka obrazovky 2024-10-12 o 17 53 04
Snímka obrazovky 2024-10-12 o 17 53 15

### CODE REVIEW

File: apple/DemoApp/Demo/DemoNavigationView.swift [hunk 1]
```diff
@@ -13,10 +13,10 @@ private let initialLocation = CLLocation(latitude: 37.332726,
                                          longitude: -122.031790)
 
 struct DemoNavigationView: View {
-    private let navigationDelegate = NavigationDelegate()
+    private var navigationDelegate = NavigationDelegate()
```
Comment (ianthetechie) [inline apple/DemoApp/Demo/DemoNavigationView.swift:16]: Does this need to be a `var`?

File: apple/DemoApp/Demo/DemoNavigationView.swift [hunk 2]
```diff
@@ -148,6 +150,23 @@ struct DemoNavigationView: View {
...
+
+                        MuteButton(isMuted: $spokenInstructionObserver.isMuted)
+
+                    }
+
+                },
+
+                alignment: .topTrailing
+
+            )
```
Comment (ianthetechie) [inline apple/DemoApp/Demo/DemoNavigationView.swift:169]: This works for the demo app, but doesn't do much for everyone building apps with the library components :)

Have a look at `PortraitNavigationOverlayView` and `LandscapeNavigationOverlayView` for how we handle this. We already built the whole grid layout and have ways to configure what's shown where, as well as overrides. Moving the mute button to the top trailing position there will also conveniently mean it's shown and hidden automatically in the future based on whether the user is navigating.

File: apple/DemoApp/Demo/MuteButtonView.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+//
+//  MuteButton.swift
+//  Ferrostar Demo
+//
+//  Created by Marek Sabol on 08/10/2024.
+//
+
```
Comment (ianthetechie) [inline apple/DemoApp/Demo/MuteButtonView.swift:7]: 1. The mute button isn't specific to the demo app but is reusable, so it should move to the `FerrostarSwiftUI` module. All similar controls are under FerrostarSwiftUI > Views > Controls.
2. We remove the automatic Xcode header comments.

File: apple/DemoApp/Demo/MuteButtonView.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 18, height: 18)
+                .padding()
+                .foregroundColor(.black)
+                .background(Color.white)
+                .clipShape(Circle())
+                .shadow(radius: 10)
+        }
+        .padding(.trailing, 18) // Right
+        .padding(.top, 112)
```
Comment (ianthetechie) [inline apple/DemoApp/Demo/MuteButtonView.swift:28]: We can probably remove most of these after integrating into the overlays. See the other controls for examlpes.

**Comment by MarekSabol** [comment]
I've opened new PR for adjusments Ios mute fix after discussion #296

**Pull request closed without merging**